airflow/utils/task_group.py

```diff
@@ -245,6 +245,8 @@ def add(self, task: DAGNode) -> None:
         elif SetupTeardownContext.is_teardown:
             if isinstance(task, AbstractOperator):
                 setattr(task, "_is_teardown", True)
+                if getattr(SetupTeardownContext, "on_failure_fail_dagrun", False):
+                    setattr(task, "_on_failure_fail_dagrun", True)
```

Comment:
If that's the case, should we still have @setup/@teardown decorator because it seems to me that users would now be able to set their tasks as setup/teardown when defining it in the task? e.g:
```python

BashOperator(task_id='mytask', is_setup=True, bash_command="echo 1")
```

Reply:
And also all tasks would now have `on_failure_fail_dagrun` making dag authors to use it on normal tasks which are not teardown tasks. cc @norm

Reply:
We can control this in the init of the BaseOperator but I doubt it's a good user experience

Reply:
The decorators would still be needed for non-classic operators, wouldn't they?

Reply:
Setting `on_failure_fail_dagrun` as a default in the BaseOperator seems weird. At least to me. Like, it should only apply to a teardown task as that's how it's described. And the default of `False` in this case is opposite to the usual behaviour of an operator of failing the run, isn't it?